server/src/main/java/org/elasticsearch/common/geo/GeoExtent.java

```diff
@@ -0,0 +1,185 @@
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.common.geo;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * An extent computes the bounds of the provided points
+ * or other extents.
+ */
+public class GeoExtent implements Writeable {
+
+    private double top;
+    private double bottom;
+    private double negLeft;
+    private double negRight;
+    private double posLeft;
+    private double posRight;
+
+    public GeoExtent() {
+        this.top = Double.NEGATIVE_INFINITY;
+        this.bottom = Double.POSITIVE_INFINITY;
+        this.negLeft = Double.POSITIVE_INFINITY;
+        this.negRight = Double.NEGATIVE_INFINITY;
+        this.posLeft = Double.POSITIVE_INFINITY;
+        this.posRight = Double.NEGATIVE_INFINITY;
+    }
+
+    public GeoExtent(StreamInput input) throws IOException {
+        this.top = input.readDouble();
+        this.bottom = input.readDouble();
+        this.negLeft = input.readDouble();
+        this.negRight = input.readDouble();
+        this.posLeft = input.readDouble();
+        this.posRight = input.readDouble();
+    }
+
+    /**
+     * Adds a point to the Extent
+     */
+    public void addPoint(double lat, double lon) {
+        this.top = Math.max(this.top, lat);
+        this.bottom = Math.min(this.bottom, lat);
+        if (lon < 0) {
+            this.negLeft = Math.min(this.negLeft, lon);
+            this.negRight = Math.max(this.negRight, lon);
+        } else {
+            this.posLeft = Math.min(this.posLeft, lon);
+            this.posRight =Math.max(this.posRight, lon);
+        }
+    }
+
+    /**
+     * Adds another extent to this extent.
+     */
+    public void addExtent(GeoExtent extent) {
+        this.top = Math.max(this.top, extent.top);
+        this.bottom = Math.min(this.bottom,  extent.bottom);
+        this.negLeft = Math.min(this.negLeft,  extent.negLeft);
+        this.negRight = Math.max(this.negRight,  extent.negRight);
+        this.posLeft = Math.min(this.posLeft,  extent.posLeft);
+        this.posRight =Math.max(this.posRight,  extent.posRight);
```

Comment:
nit: unbalanced spaces around equals, and double space on second argument for min/max